### ISSUE TITLE
Korean Numeral Units Issue

### ISSUE THREAD

**Issue by jsung1014** [wordplaydev/wordplay#448]
## Expected behavior

- When typing Korean numeral units in the Wordplay, we are expecting the characters should automatically combine to form the correct Hangul syllables. For instance, when typing the unit “개"  using its constituent parts, the input method should seamlessly combine “ㄱ” and “ㅐ" into the single syllable “개" without displaying them as separate components “3 ㄱ ㅐ". 

## Actual behavior

- However, when typing Korean numeral units in the current WordplayPen, the characters do not combine as expected. Instead of forming the correct, the input sequence breaks down. For example, when we typed “개" as a Korean numeral unit, it broke down like “ㄱ ㅐ".

Provide detailed steps that consistently reproduce the behavior, including any example code. If you can, make the steps as minimal as possible, finding the smallest possible set of steps that still produce the unexpected behavior.

1. Download the Korean keyboard(2-Set Korean) on your laptop or ask for help from a Korean student to follow this process. 
2. Go to the Wordplay website by searching on Google from your laptop.
3. Click on the “Learning” button to open the page. 
4. Click on the documentation icon to open the text editing area.
5. Click into the text input area to position the cursor where you want to start typing.
6. Type the numeral "3", followed by the Korean components for the syllable "개":
7. Press the key for "ㄱ". (Same as the “R” key in English)
8. Press the key for "ㅐ". (Same as the “O” key in English)
9. Checking the error if the characters are incorrectly displayed as separate entities like “3 ㄱ ㅐ".

## Screenshots
<img width="1440" alt="image" src="https://github.com/wordplaydev/wordplay/assets/114701651/73ad9b5f-d436-47a8-899b-1d11c732c5dc">

## Environment

What platform, OS, browser, and version are you using? Please be exact with version numbers, as different versions can result in a defect happening or not. Don't skip these; cross-browser issues are a common cause of problems.

Laptop

-   OS + version: Mac Version 13.2.1 
-   Browser + version: Chrome 124.0.6367.60
-   Keyboard Input setting: U.S. and 2-Set Korean

## Design specification

_If this defect requires some redesign, include a design proposal here (and if you don't have one, add the `needs design` label, and you can just write TBD here). Design proposals should describe, in precise detail \_what_ the proposed design is, but not necessarily how it might be built. Note that the design proposal for a defect should align with the expected behavior you described above. Once the design is approved, we will add the `buildable` tag is added.\_

**Comment by amyjko** [comment]
Thanks for reporting! One clarifying question: what input mode and keyboard setting are you using on macOS? (e.g., U.S., 2-Set Korean). This determines how input is handled in the browser, so this will be important context for design and fixing.

**Comment by jsung1014** [comment]
@amyjko Thanks for the feedback! I just edited the bug report to update the specific keyboard setting.

**Comment by anhourandaquarter** [comment]
**Context:** I know we've been discussing fonts and possibly a font-based reason as to why the characters aren't combining properly. @amyjko showed me where fonts live in other locales, like Chinese and English. I think this means we'll have to go into the existing korean-locale branch and see what we want to do with it (merging conflicts, rebasing, or some combination of the two?).

**Question:** Should we open a new issue to handle updating the korean-locale branch? Or do we want to update the branch as part of this issue?

**Comment by amyjko** [comment]
@anhourandaquarter This issue is probably related to text input in the editor (see `Editor.svelte`), so I wouldn't expand the scope to include the whole locale. If there's not an issue for a Korean locale, open a new one for that, so we can track progress on it, and which branch it lives in.

**Comment by jsung1014** [comment]
Here is my suggestion for further process: 

Verify Keyboard Input Method:

Test the keyboard input in another text editor (e.g., Notepad, Word) to confirm that it correctly combines Korean characters outside of the WordplayPen environment.

Browser Compatibility:

Check if the issue persists across different web browsers (e.g., Chrome, Firefox, Edge).
Clear the cache and cookies of the browser to eliminate any potential conflicts.

Check WordplayPen Settings:

Look into the WordplayPen documentation or settings to see if there are any options related to text input methods, language settings, or special character support.
Ensure that the text input area is configured to support Unicode or complex script input.

Research better font: 

Validate the assumption that font might be the problem of this issue. If so, find a new font that works better than the current one. 

Developer Tools Inspection (Recommend to collaborate with Developers):

Use the browser's developer tools (usually accessed by pressing F12) to inspect the input area and check for any JavaScript errors or console messages that might indicate why the characters are not combining correctly.
Look at the HTML and CSS properties of the text input area to ensure there are no overriding styles or scripts affecting the input behavior.

**Comment by jsung1014** [comment]
After that, I would begin to limit the number of Korean numeral units that will be used in wordplay. 

In order to do so, you need to understand the Korean numeral unit system and identify a certain number of Korean numeral units. 

After that, collaborate with developers for further steps.

**Comment by amyjko** [comment]
Thank you @jsung1014!

**Comment by amyjko** [comment]
It turns out this was due to an incorrectly named Korean font file. This was resolved by commit d15ea53.
